File: VB-Examples/29_Print/PrintWithoutDialog.vb
Imports System.Text
Imports System.IO
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports System.Drawing.Printing

Namespace PrintWithoutDialog
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Specify the input file path
			Dim input As String = "..\..\..\..\..\..\Data\Sample.docx"

			' Create a new instance of Document
			Dim doc As New Document()

			' Load the Word document from the specified input file
			doc.LoadFromFile(input)

			' Get the PrintDocument associated with the document
			Dim printDoc As PrintDocument = doc.PrintDocument

			' Set the print controller to StandardPrintController for silent printing
			printDoc.PrintController = New StandardPrintController()

			' Print the document
			printDoc.Print()

			' Dispose of the document object when finished using it
			doc.Dispose()
		End Sub
	End Class
End Namespace

File: CS-Examples/29_Print/PrintWithoutDialog.cs
using System;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.IO;
using Spire.Doc;
using Spire.Doc.Documents;
using System.Drawing.Printing;

namespace PrintWithoutDialog
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
          
            // Specify the input file path
            String input = @"..\..\..\..\..\..\Data\Sample.docx";

            // Create a new instance of Document
            Document doc = new Document();

            // Load the Word document from the specified input file
            doc.LoadFromFile(input);

            // Get the PrintDocument associated with the document
            PrintDocument printDoc = doc.PrintDocument;

            // Set the print controller to StandardPrintController for silent printing
            printDoc.PrintController = new StandardPrintController();

            // Print the document
            printDoc.Print();

            // Dispose of the document object when finished using it
            doc.Dispose();
        }
    }
}
